scripts/setup-roles.ts

```diff
@@ -0,0 +1,89 @@
+import { ethers } from "hardhat";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import { ORIGINATOR_ROLE as DEFAULT_ORIGINATOR_ROLE, ADMIN_ROLE as DEFAULT_ADMIN_ROLE, FEE_CLAIMER_ROLE as DEFAULT_FEE_CLAIMER_ROLE } from "./utils/constants";
+
+export async function main (
+    VAULT_FACTORY_ADDRESS = "0x3A54241cB7801BDea625565AAcb0e873e79C0649",
+    LENDER_NOTE_ADDRESS = "0xdeaBbBe620EDF275F06E75E8fab18183389d606F",
+    BORROWER_NOTE_ADDRESS = "0xaEF48370a5f37CFb760CE44E6cbF986C4DeFF389",
+    ORIGINATION_CONTROLLER_ADDRESS = "0xFE046149bc8830d989A007E81A9D7F8A3b575F36",
+    LOAN_CORE_ADDRESS = "0xd624D1879429A606f54F48B08b56126c3Fe70049",
+    ADMIN_ADDRESS = "0x9b419fd36837558D8A3197a28a5e580AcE44f64F",
+    FEE_CONTROLLER_ADDRESS = "0xE4a1917Ebe8fd2CAFD79799C82aDAa7E81AC6D47",
+    ADMIN_ROLE = DEFAULT_ADMIN_ROLE,
+    FEE_CLAIMER_ROLE = DEFAULT_FEE_CLAIMER_ROLE,
+    ORIGINATOR_ROLE = DEFAULT_ORIGINATOR_ROLE,
+): Promise<void> {
+    const signers: SignerWithAddress[] = await ethers.getSigners();
+    const [admin, borrower] = signers;
+    const deployer = admin;
+
+    if (!LOAN_CORE_ADDRESS) {
+        throw new Error("Must specify LOAN_CORE_ADDRESS in environment!");
+    }
+    if (!admin.address) {
+        throw new Error("Must specify ADMIN_ADDRESS in environment!");
+    }
+    if (FEE_CONTROLLER_ADDRESS) {
+        console.log(`Fee controller address: ${FEE_CONTROLLER_ADDRESS}`);
+    }
+
+    const loanCore = await ethers.getContractAt("LoanCore", `${LOAN_CORE_ADDRESS}`);
+    const lenderNote = await ethers.getContractAt("PromissoryNote", `${LENDER_NOTE_ADDRESS}`);
+    const borrowerNote = await ethers.getContractAt("PromissoryNote", `${BORROWER_NOTE_ADDRESS}`);
+    const vaultFactory = await ethers.getContractAt("VaultFactory", `${VAULT_FACTORY_ADDRESS}`);
+
+    //Grant correct permissions for promissory note
+    //Giving to user to call PromissoryNote functions directly
+    for (const note of [borrowerNote, lenderNote]) {
+        await note.connect(await admin).initialize(loanCore.address);
+    }
+
+    // grant the borrower: FEE_CLAIMER_ROLE
+    const updateBorrowerPermissions = await loanCore.connect(admin).grantRole(FEE_CLAIMER_ROLE, borrower.address)
```

Comment:
I think if you remove the liens above and make the final argument here `admin.address` you're good

Reply:
We need lines 39 - 43 to to initialize loanCore for promissory note to avoid this:
![Screen Shot 2022-05-25 at 9 03 09 PM](https://user-images.githubusercontent.com/42501101/170393852-ebfa6381-4fc5-41d2-b294-2e2ef5a225a9.png)